src/mongocxx/exception/exception.cpp

```diff
@@ -10,29 +10,22 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
-// limitations under the License./**
+// limitations under the License.
 
 #include <mongocxx/config/prelude.hpp>
 
-#include <cstdint>
 #include <string>
+#include <utility>
 
-// TODO: figure out if we need this
-// TODO: figure out if bson document should be view/value, etc.
+#include <mongocxx/exception/exception.hpp>
 
 namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
-namespace error {
+namespace exception {
```

Comment:
The advantage of the exception namespace is that the derived classes read mongocxx::exception::operation, etc (and in the case of bulk_write, this avoids a clash with the existing mongocxx::bulk_write). But I'm happy to call the derived classes operation_exception, bulk_write_exception, etc.